### PR TITLE
add is_torch_min_version in fsdp src

### PR DESCRIPTION
**Pull request by xrennvidia**
# What does this PR do ?
To fix this [FSDP wheel building issue](https://github.com/NVIDIA/Megatron-LM/actions/runs/25835932209/job/76055062199).

## Issue tracking

For PRs from open-source community contributors:

- **New features**: a linked issue is **required**. Please open a [feature request](https://github.com/NVIDIA/Megatron-LM/issues/new?template=feature_request.md) and reference it here before submitting the PR.
- **Small updates (bug fixes, minor improvements)**: a linked issue is **recommended** and will accelerate the PR review process.

Linked issue: 

### Pre-checks

- [ ] I have added relevant unit tests
- [ ] I have added relevant functional tests
- [ ] I have added proper typing to my code [Typing guidelines](https://docs.python.org/3/library/typing.html)
- [ ] I have added relevant documentation
- [ ] I have run the [autoformatter.sh](https://github.com/NVIDIA/Megatron-LM/blob/main/tools/autoformat.sh) on my PR

### Code review

Feel free to message or comment the [@mcore-oncall](https://github.com/orgs/NVIDIA/teams/mcore-oncall) to help accelerate your merge into main. The less complex your PR is, the faster it will be approved and merged!

All PRs start as **draft**. If you open a non-draft PR, it will be automatically converted to draft.

#### Step 1: Mark PR as "Ready for Review"

1. When your PR is ready, click **Ready for Review**.
2. An oncall reviewer is auto-assigned and expert reviewers are notified based on your changes.
   - Some PRs may jump straight to step 2. This is determined by `.github/CODEOWNERS`.

:warning: Only mark as ready once merge-conflicts are resolved and the CI is passing.
Final Review might get declined if these requirements are not fulfilled.

#### Step 2: Final Review

For PRs that change `megatron/core`, once all expert reviewers have approved, the `Final Review` label is applied **automatically** and final reviewers are assigned.

For PRs outside `megatron/core`, this step is skipped.

#### Step 3: Approved

Once all required reviewers have approved, the `Approved` label is applied **automatically**.

### Merge

Any member of [mcore-engineers](https://github.com/orgs/NVIDIA/teams/mcore-engineers) will be able to merge your PR.

<details>
<summary>For MRs into `dev` branch</summary>
The proposed review process for `dev` branch is under active discussion.

MRs are mergable after one approval by either `eharper@nvidia.com` or `zijiey@nvidia.com`.
</details>

### CODE REVIEW

**Review by chtruong814 — APPROVED** [review]

**Review by cspades — APPROVED** [review]
LGTM, I made a similar change in my CG PR where I will remove the MCore import from DDPConfig but I like this better, just have a util for it.

File: megatron/core/distributed/fsdp/src/megatron_fsdp/distributed_data_parallel_config.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 import torch
 
-from megatron.core.utils import is_torch_min_version
+from .utils import is_torch_min_version
```
Comment (ericharper) [inline megatron/core/distributed/fsdp/src/megatron_fsdp/distributed_data_parallel_config.py:8]: can we use a full path?

File: megatron/core/distributed/fsdp/src/megatron_fsdp/utils.py [hunk 2]
```diff
@@ -46,6 +46,13 @@
     HAVE_TE = False
 
 
+try:
+    _torch_version = PkgVersion(torch.__version__)
+except Exception:
+    # This is a WAR for building docs, where torch is not actually imported
+    _torch_version = PkgVersion("0.0.0")
```
Comment (jaredcasper) [inline megatron/core/distributed/fsdp/src/megatron_fsdp/utils.py:53]: Will this lead to weird error messages where it says torch is too old when in fact it isn't even installed?

File: megatron/core/distributed/fsdp/src/megatron_fsdp/distributed_data_parallel_config.py (hunk 1 above)
Comment (cspades) [reply megatron/core/distributed/fsdp/src/megatron_fsdp/distributed_data_parallel_config.py:8]: There's two possible paths depending on the install:

```
megatron.core.distributed.fsdp.src.megatron_fsdp.utils
megatron_fsdp.utils
```

**Review by yaoyu-33 — APPROVED** [review]

**Review by jaredcasper — APPROVED** [review]
